### PR TITLE
Use Engine Version From Config

### PR DESCRIPTION
**Pull request by Fausto95**
Use engine version from /config

### CODE REVIEW

**Comment by Fausto95** [comment]
[As Mobiliar, i'd like to have 3 lives on mobile app](https://trello.com/c/lLw8Gy04/1399-as-mobiliar-id-like-to-have-3-lives-on-mobile-app)

File: src/types.js [hunk 1]
```diff
@@ -82,6 +82,13 @@ export type Section = {|
   cardsRef?: Array<string | void>
 |};
 
+export type ProgressionEngineVersions = {|
+  versions: {
+    learner: string,
+    microlearning: string
```
Comment (VincentCATILLON) [inline src/types.js]:
```
versions: {
  [Engine]?: string
}
```

File: src/redux/actions/progressions/create-next-progressions.test.js [hunk 2]
```diff
@@ -19,6 +19,19 @@ const level = createLevel({
...
+    brand: {
+      progressionEngine: {
+        versions: {
+          learner: '2',
+          microlearning: '2'
+        }
+      }
+    }
+  }
+});
```
Comment (VincentCATILLON) [inline src/redux/actions/progressions/create-next-progressions.test.js]: Use store fixture for this please:
- Add a `createAuthenticationState` in `src/__fixtures__/store.js`
- Use it by default like `catalog`: https://github.com/CoorpAcademy/mobile/blob/master/src/__fixtures__/store.js#L214
- Create store state in this test file

File: src/redux/actions/progressions/create-next-progression.js [hunk 3]
```diff
@@ -45,6 +45,12 @@ export const createNextProgression = (
       throw new Error(`content type ${type} is not handled`);
     }
 
+    const {
+      authentication: {
+        brand: {progressionEngine}
```
Comment (VincentCATILLON) [inline src/redux/actions/progressions/create-next-progression.js]: `progressionEngine` is unknown in `undefined` (brand can be undefined)

File: src/redux/actions/progressions/create-level-progression.js [hunk 4]
```diff
@@ -8,16 +8,13 @@ import {ObjectId} from 'bson';
 import {ENGINE} from '../../../const';
 import {getMostAccurateRef} from '../../../modules/reference';
 
-const ENGINE_VERSION = '2';
-const ENGINE_CONFIG_VERSION = '2';
-
-export const createLevelProgression = (level: Level) => {
+export const createLevelProgression = (level: Level, enginesVersion: string) => {
```
Comment (VincentCATILLON) [inline src/redux/actions/progressions/create-level-progression.js]: please rename to `version` and type it optional (as the API may not return it), so default value should be the constant value

File: src/redux/actions/progressions/create-chapter-progression.js [hunk 5]
```diff
@@ -8,14 +8,11 @@ import {ObjectId} from 'bson';
 import {getMostAccurateRef} from '../../../modules/reference';
 import {ENGINE} from '../../../const';
 
-const ENGINE_VERSION = '2';
-const ENGINE_CONFIG_VERSION = '2';
-
-export const createChapterProgression = (chapter: Chapter) => {
-  const engine: Engine = {ref: ENGINE.MICROLEARNING, version: ENGINE_VERSION};
+export const createChapterProgression = (chapter: Chapter, enginesVersion: string) => {
```
Comment (VincentCATILLON) [inline src/redux/actions/progressions/create-chapter-progression.js]: please rename to `version` and type it optional (as the API may not return it), so default value should be the constant value

File: src/__fixtures__/brands.js [hunk 6]
```diff
@@ -12,5 +12,11 @@ export const createBrand = (): Brand => ({
   images: {
     'logo-mobile':
       'https://static.coorpacademy.com/content/mobile/raw/coorp_logo_infinite-1552063832916.png'
+  },
```
Comment (VincentCATILLON) [inline src/__fixtures__/brands.js]: please add `progressionEngine` in function params to be used in `createAuthenticationState` (in `__fixtures__/store`)

File: src/__fixtures__/store.js [hunk 7]
```diff
@@ -211,7 +214,7 @@ export const createStoreState = ({
     },
     catalog: catalog || createCatalogState(),
     permissions: permissionsState,
-    authentication,
+    authentication: authentication || createAuthenticationState({}),
```
Comment (VincentCATILLON) [inline src/__fixtures__/store.js]: :prie:

File: src/components/brand-theme-provider.js [hunk 8]
```diff
@@ -41,6 +41,8 @@ const BrandThemeProvider = ({children, brand}: Props) => (
   <BrandThemeContext.Provider value={brand}>{children}</BrandThemeContext.Provider>
 );
 
+// ignore this line since its for codecov, because the brand is no longer null in the tests
+/* istanbul ignore next */
```
Comment (VincentCATILLON) [inline src/components/brand-theme-provider.js]: no go for it

File: src/redux/actions/progressions/create-next-progression.js [hunk 9]
```diff
@@ -45,11 +45,12 @@ export const createNextProgression = (
...
-      authentication: {
-        brand: {progressionEngine}
-      }
-    } = getState();
+    const {authentication} = getState();
+    const engineVersions =
+      authentication &&
+      authentication.brand &&
+      authentication.brand.progressionEngine &&
+      authentication.brand.progressionEngine.versions;
```
Comment (VincentCATILLON) [inline src/redux/actions/progressions/create-next-progression.js]: can you use a `getEngineVersion()` state extractor ?

File: src/redux/actions/progressions/create-next-progressions.test.js [hunk 10]
```diff
@@ -19,19 +21,24 @@ const level = createLevel({
...
 
+const getState = () =>
+  createStoreState({
+    levels: [],
+    disciplines: [],
+    chapters: [],
+    slides: [],
+    progression,
+    authentication: createAuthenticationState({})
+  });
```
Comment (VincentCATILLON) [inline src/redux/actions/progressions/create-next-progressions.test.js]: nice one dude

File: src/types.js [hunk 11]
```diff
@@ -84,8 +84,7 @@ export type Section = {|
 
 export type ProgressionEngineVersions = {|
   versions: {
-    learner: string,
-    microlearning: string
+    [Engine]: string
```
Comment (VincentCATILLON) [inline src/types.js]: same for this 👍

File: src/redux/utils/state-extract.js [hunk 12]
```diff
@@ -123,3 +123,10 @@ export const getSection = (state: StoreState, key: string): Section | void =>
 export const getCard = (state: StoreState, ref: string): DisciplineCard | ChapterCard | void =>
   state.catalog.entities.cards[ref] &&
   state.catalog.entities.cards[ref][translations.getLanguage()];
+
+export const getEngineVersions = (state: StoreState) =>
```
Comment (VincentCATILLON) [inline src/redux/utils/state-extract.js]: return type please

**Pull request merged**